aptos-move/framework/supra-framework/sources/multisig_voting.spec.move

```diff
@@ -0,0 +1,427 @@
+        ensures len(next_execution_hash) == 0 ==> post_proposal.is_resolved == true;
+        ensures (len(next_execution_hash) == 0 && is_multi_step) ==> simple_map::spec_get(post_proposal.metadata, multi_step_in_execution_key) == std::bcs::serialize(false);
+        // property 4: For multi-step proposals, it ensures that the next_execution_hash parameter contains the hash of the next step.
+        ensures len(next_execution_hash) != 0 ==> post_proposal.execution_hash == next_execution_hash;
+    }
+
+    spec next_proposal_id {
+        aborts_if !exists<VotingForum<ProposalType>>(voting_forum_address);
+        ensures result == global<VotingForum<ProposalType>>(voting_forum_address).next_proposal_id;
+    }
+
+    spec get_proposer {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.proposer;
+    }
+
+    spec is_voting_closed {
+        use supra_framework::chain_status;
+        // Ensures existence of Timestamp
+        requires chain_status::is_operating();
+        //TODO: Remove pragma aborts_if_is_partial;
+        pragma aborts_if_is_partial = true;
+        include AbortsIfNotContainProposalID<ProposalType>;
+        // aborts_if !exists<timestamp::CurrentTimeMicroseconds>(@supra_framework);
+        // ensures result == spec_is_voting_closed<ProposalType>(voting_forum_address, proposal_id);
+    }
+
+    spec can_be_resolved_early {
+        //TODO: Remove pragma aborts_if_is_partial;
+        pragma aborts_if_is_partial = true;
+        ensures result == spec_can_be_resolved_early<ProposalType>(proposal);
+    }
+
+    spec get_proposal_metadata {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.metadata;
+    }
+
+    spec get_proposal_metadata_value {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        aborts_if !simple_map::spec_contains_key(proposal.metadata, metadata_key);
+        ensures result == simple_map::spec_get(proposal.metadata, metadata_key);
+    }
+
+    spec get_proposal_state {
+        use supra_framework::chain_status;
+
+        pragma addition_overflow_unchecked;
+        // Ensures existence of Timestamp
+        requires chain_status::is_operating();
+        //TODO: Remove pragma aborts_if_is_partial;
+        pragma aborts_if_is_partial = true;
+        include AbortsIfNotContainProposalID<ProposalType>;
+
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
```

Comment:
No property here either, it's commented out, why?

Reply:
Fixed in [93c26ef](https://github.com/Entropy-Foundation/aptos-core/pull/63/commits/93c26ef7e98251f03f9205ec306217d00d0a82e5).